core/src/main/java/org/wildfly/channel/ChannelImpl.java

```diff
@@ -194,12 +205,92 @@ static class ResolveLatestVersionResult {
     }
 
     private ChannelManifest resolveManifest(ChannelManifestCoordinate manifestCoordinate) throws UnresolvedMavenArtifactException {
-        return resolver.resolveChannelMetadata(List.of(manifestCoordinate))
+        return resolveChannelMetadata(List.of(manifestCoordinate), false)
                 .stream()
                 .map(ChannelManifestMapper::from)
                 .findFirst().orElseThrow();
     }
 
+    /**
+     * Resolve a list of channel metadata artifacts based on the coordinates.
+     * If the {@code ChannelMetadataCoordinate} contains non-null URL, that URL is returned.
+     * If the {@code ChannelMetadataCoordinate} contains non-null Maven coordinates, the Maven artifact will be resolved
+     * and a URL to it will be returned.
+     * If the Maven coordinates specify only groupId and artifactId, latest available version of matching Maven artifact
+     * will be resolved.
+     *
+     * The order of returned URLs is the same as order of coordinates.
+     *
+     * @param coords - list of ChannelMetadataCoordinate.
+     * @param optional - if artifact is optional, the method will return an empty collection if no versions are found
```

Comment:
it's not clear to me what "if artifact is optional" means? When would an artifact be optional?